src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/GraphNodes/DependenciesGraphProvider.cs

```diff
@@ -132,9 +132,14 @@ internal async Task BeginGetGraphDataAsync(IGraphContext context)
             {
                 await InitializeAsync();
 
-                IEnumerable<Lazy<IDependenciesGraphActionHandler, IOrderPrecedenceMetadataView>> actionHandlers = GraphActionHandlers.Where(x => x.Value.CanHandleRequest(context));
-                bool shouldTrackChanges = actionHandlers.Aggregate(
-                    false, (previousTrackFlag, handler) => previousTrackFlag || handler.Value.HandleRequest(context));
+                bool shouldTrackChanges = false;
+                foreach (Lazy<IDependenciesGraphActionHandler, IOrderPrecedenceMetadataView> handler in GraphActionHandlers)
+                {
+                    if (handler.Value.CanHandleRequest(context))
+                    {
+                        shouldTrackChanges = shouldTrackChanges || handler.Value.HandleRequest(context);
```

Comment:
Can't break early, but the `if` statement's easier to read. The `b = b || c` form came from unravelling the `Aggregate` code. I'll update to use `if`.

Reply:
Hang on, it's early. I take it back -- we can break early.

Reply:
You should take it as a compliment that I double checked a dozen times in my head that an early exit was possible, because you seem to rarely do anything without a good reason.

Reply:
Flattery will get you beer when you're in London.